policies/ip-policy.md

```diff
@@ -3,51 +3,77 @@ status: draft
 
-Contributions are validated through a [Developer Certificate of Origin][DCO] (DCO) or a Contributor License Agreement (CLA), as determined by the project.
+## Contributions
 
-### Documentation and Media Files
+> [!TIP]
+> All code contributions must align with the project’s license and come with a promise (DCO) from contributors that they have the right to contribute under these terms.
 
-Projects should choose an appropriate license for documentation and media files, such as [CC BY 4.0][] or the project's primary license.
+All new code contributions to any Project shall be made under the Project Code License accompanied by a Developers Certificate of Origin (DCO, available at <http://developercertificate.org/>), which will bind the individual contributor and, if applicable, their employer to the Project Code License.
 
-All contributions must be clearly labeled with license information.
+The technical governing body of each Project may provide for additional requirements with respect to contributions. For example, a Project may require that new code contributions to a Project be accompanied by a signed Contributor License Agreement (CLA).
 
-## Collaborations and External Projects
+### Developers Certificate of Origin (DCO) or Contributor License Agreement (CLA)?
+
+The DCO and CLA differ in their approach to intellectual property rights.
 
-When collaborating with external open-source projects ("Upstream Projects"):
+- A DCO is a lightweight method for contributors to certify they have the rights to submit their work, typically requiring a signed statement within the commit message.
+- A CLA is a more formal agreement where contributors grant the project rights to use their contributions, often including a detailed legal framework around contributions.
 
-- Comply with the Upstream Project's licensing requirements.
-- Prioritize compatibility
-- Seek alternative license approvals from the CFC when necessary.
+CLAs can provide projects with more legal protection, while DCOs offer a simpler, less legally invasive way for developers to contribute.
 
 ## Intellectual Property Rights
 
-CF membership does not grant or transfer intellectual property rights unless explicitly stated by this policy or a Project's license.
+> [!TIP]
+> Joining CF doesn’t grant us any rights to your IP beyond what’s needed for project contributions. Your IP remains yours.
+
+Except for (a) each member’s commitment to be bound by this Policy, the applicable Project Code License(s), and the applicable Project Documentation License(s), in each case with respect to its or its employees’ authorized contributions to any Project; and (b) any applicable contributor license agreement, if any, no license is granted by the member to its intellectual property, and none shall be implied by general membership in the Commonhaus Foundation.
```

Comment:
I'm not entirely sure where (b) ends to indicate what the IP rights are which excludes (a) and (b) situations. I can presume it's after ", if any, ", but I'm not 100% sure. Do we need a semi-colon at the end of (b) like (a) has?

Reply:
you don't usually have a semi-colon at list end. But yea.. that sentence is a mouthful.

An inverted structure could work. Is this better? (I think it covers all the bases..)

> General membership in the Commonhaus Foundation does not imply any license to a member's intellectual property. Members grant no license to their intellectual property to the Commonhaus Foundation, except under the following conditions: (a) their commitment to abide by this Policy, the applicable Project Code License(s), and the applicable Project Documentation License(s) for their and their employees' contributions to any Project; and (b) any applicable contributor license agreement.

Reply:
Added.

Reply:
I like this better and don't think anything was dropped